### PR TITLE
Add an SSL BIO test for QUIC

### PR DESCRIPTION
**Pull request by mattcaswell**
We create an SSL BIO using a QUIC based SSL_CTX and then use that BIO to create a connection and read/write data from streams.

### CODE REVIEW

File: test/quicapitest.c [hunk 1]
```diff
@@ -578,6 +578,138 @@ static int test_quic_set_fd(int idx)
...
+        SSL_handle_events(clientquic);
+        if (!TEST_true(BIO_read_ex(thisbio, buf, sizeof(buf), &readbytes))
+                || !TEST_mem_eq(msg, msglen, buf, readbytes))
+            goto err;
+
+        if (i == 1)
+            break;
+
+        /* Now create a new stream and repeat */
+        sid = 4;
```
Comment (paulidale) [inline test/quicapitest.c:683]: Comment explaining this _magic_ number perhaps?

**Review by paulidale — COMMENTED** [review]
CI is relevant.

**Comment by mattcaswell** [comment]
Updates pushed to resolve the CI failure and to address the feedback. Please take another look.

**Comment by mattcaswell** [comment]
The remaining CI failure is the multistream (script 18) failure which has been seen elsewhere.

**Review by paulidale — APPROVED** [review]

**Comment by mattcaswell** [comment]
Once #21381 has merged I can remove the TODO comment from this PR and rebase it. Aside from that this is still looking for a second reviewer.

**Comment by paulidale** [comment]
Approved without the TODO as well :grin:

**Review by hlandau — APPROVED** [review]

**Comment by mattcaswell** [comment]
Updated to remove the TODO and associated call to `SSL_handle_events()`. I assume @paulidale's approval still holds since he explicitly mentioned this.

@hlandau - please reconfirm

**Review by hlandau — APPROVED** [review]

**Pull request closed without merging**